doc/shared/sunstepper/SUNStepper_Description.rst

```diff
@@ -0,0 +1,348 @@
+
+   This function takes a single internal step starting at *t0* towards *tout*
+   and stores the next step solution in *vout*.
+
+   :param stepper: the stepper object.
+   :param t0: the initial time for the integration.
+   :param tout: the final time for the integration.
+   :param vout: on output, the state after a single internal step.
+   :param tret: the final time corresponding to the output value *vout*.
+   :return: A :c:type:`SUNErrCode` indicating success or failure.
+
+
+.. c:function:: SUNErrCode SUNStepper_Reset(SUNStepper stepper, sunrealtype tR, N_Vector vR)
+
+   This function resets the stepper state to the provided independent variable
+   value and dependent variable vector.
+
+   :param stepper: the stepper object.
+   :param tR: the value of the independent variable :math:`t_R`.
+   :param vR: the value of the dependent variable vector :math:`v(t_R)`.
+   :return: A :c:type:`SUNErrCode` indicating success or failure.
+
+
+.. c:function:: SUNErrCode SUNStepper_SetStopTime(SUNStepper stepper, sunrealtype tstop)
+
+   This function specifies the value of the independent variable :math:`t` past
+   which the solution is not to proceed.
+
+   :param stepper: the stepper object.
+   :param tstop: stopping time for the stepper.
+   :return: A :c:type:`SUNErrCode` indicating success or failure.
+
+
+.. c:function:: SUNErrCode SUNStepper_SetForcing(SUNStepper stepper, sunrealtype tshift, sunrealtype tscale, N_Vector* forcing, int nforcing)
+   
+   This function sets the data necessary to compute the forcing term
+   :eq:`SUNStepper_forcing`. This includes the shift and scaling factors for the
+   normalized time :math:`\frac{t - t_{\text{shift}}}{t_{\text{scale}}}` and the
+   array of polynomial coefficient vectors :math:`\widehat{f}_k`.
+
+   :param stepper: a stepper object.
+   :param tshift: the time shift to apply to the current time when computing
+      the forcing, :math:`t_{\text{shift}}`.
+   :param tscale: the time scaling to apply to the current time when computing
+      the forcing, :math:`t_{\text{scale}}`.
+   :param forcing: a pointer to an array of forcing vectors,
+      :math:`\widehat{f}_k`.
+   :param nforcing: the number of forcing vectors, :math:`n_{\text{forcing}}`. A
+      value of 0 effectively eliminates the forcing term.
+   :return: A :c:type:`SUNErrCode` indicating success or failure.
+
+   .. note::
+
+      When integrating the ODE :eq:`SUNStepper_IVP` the :c:type:`SUNStepper` is
+      responsible for evaluating ODE right-hand side function :math:`f(t, v)` as
+      well as computing and applying the forcing term :eq:`SUNStepper_forcing`
+      to obtain the full right-hand side of the ODE :eq:`SUNStepper_IVP`.
+
+.. _SUNStepper.Description.BaseMethods.Content:
+
+Attaching and Accessing the Content Pointer
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+.. c:function:: SUNErrCode SUNStepper_SetContent(SUNStepper stepper, void *content)
```

Comment:
Note for reviewers: these functions are new to `SUNStepper` compared to `MRIStepInnerStepper`. Since the `SUNStepper` struct is private, I think this is the proper way to access the `last_flag`.